components/sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link_icon.jsx

```diff
@@ -25,6 +29,21 @@ export default class SidebarChannelButtonOrLinkIcon extends React.PureComponent
         teammateIsBot: PropTypes.bool,
     };
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            svgError: false,
+            botIconUrl: null,
+        };
+    }
+
+    onSvgLoadError = () => {
+        this.setState({
+            svgError: true,
+            botIconUrl: this.props.botIconUrl,
```

Comment:
do we need to keep track of this? can't we always use `this.props.botIconUrl`?

Reply:
@renilJoseph should this be removed?

Reply:
@ali-farooq0 because, based on my understanding that state values will be preserved. So say next time page is loaded and got botIconUrl as null from server, then `this.props.botIconUrl` will be null. But `this.state.botIconUrl` will have the correct url which was saved last time and use this in SVG.

Reply:
@renilJoseph I think in the case where `this.props.botIconUrl` is `null` we want to show the default icon, not the icon that was saved from the last time 🤔

Reply:
@ali-farooq0 Thank you for that catch. I reverted the logic as exactly as it was in original branch: `https://github.com/mattermost/mattermost-webapp/pull/3311` and tested it.

Reply:
Thanks @renilJoseph!! I think you can even go one step-further from what I had in https://github.com/mattermost/mattermost-webapp/pull/3311 by removing `this.state.botIconUrl` altogether. As in my change I _always_ had to fetch the boticon and if the fetch failed i reverted it back to the default icon. The good thing about your change is that if `this.props.botIconUrl` is null that means you don't need to fetch it.

so i think you can just do something like
```
                if (this.props.botIconUrl &&
                    this.props.botIconUrl !== this.state.svgErrorUrl) {
                    icon = (
                        <Svg
                            className='icon icon__bot'
                            src={this.props.botIconUrl}
                            onLoad={this.onSvgLoad}
                            onError={this.onSvgLoadError}
                        />
                    );
                }
```

Reply:
Really appreciate your patience on working through this @renilJoseph 🥇 💯 :) I think in the end the code will be a lot simpler :)

Reply:
@ali-farooq0 Thanks for your patience in helping me. It really feels good to do some contribution. Did the change.